### PR TITLE
feat: open in browser and proxy paths

### PR DESCRIPTION
**Pull request by sastan**
- "open in browser": Opens the dev server in a new browser tab.
- "proxy paths": Define different paths that should be proxied, and where they should be proxied to.

PS: This is a great library. Thank you! This helps us a lot!

### CODE REVIEW

**Comment by sastan** [comment]
Is there anything i can do to help merge this MR?

File: lib/hmr.js [hunk 1]
```diff
@@ -317,12 +322,14 @@ module.exports = (options = {}) => {
...
-      return url
-    })
+    const forget = Object.keys(changed)
+      .map(file => {
+        const url = fileUrls[file]
+        delete fileUrls[file]
+        delete changed[file]
+        return url
+      })
+      .filter(Boolean)
```
Comment (rixo) [inline lib/hmr.js:332]: Why? Is this fixing an existing bug? Something new?

File: lib/server.js [hunk 2]
```diff
@@ -140,6 +141,7 @@ const createServer = ({
   const modules = {}
 
   const writeBundle = (baseUrl, bundle, { sourcemap } = {}) => {
+    console.time('server.writeBundle')
```
Comment (rixo) [inline lib/server.js:144]: Do we need this? Seems very verbose (log wise) to me.

**Comment by rixo** [comment]
Sorry, I've been working on a very intense mission and barely had time to breath these past weeks... Thanks for your contribution :)

Can we split this into a PR for opening browser and another one for proxy?

I think I'd prefer to keep default to opening no browser for now.

I'd appreciate if you added some licence info in the code copied from Snowpack.

I've added a few other questions in the comments.

**Comment by sastan** [comment]
No worries. I will split this MR up and incorporate your comments.

File: lib/hmr.js (hunk 1 above)
Comment (sastan) [reply lib/hmr.js:332]: See #14

File: lib/server.js (hunk 2 above)
Comment (sastan) [reply lib/server.js:144]: Forgot to remove that.

**Comment by sastan** [comment]
I have created separate PRs for each feature/fix:

- #12 
- #13 
- #14

**Pull request closed without merging**